internal/pkg/template/template.go

```diff
@@ -63,6 +99,49 @@ func (t *Template) Parse(path string, data interface{}, options ...ParseOption)
 	return &Content{buf}, nil
 }
 
+// UploadEnvironmentCustomResources uploads the environment custom resource scripts.
+func (t *Template) UploadEnvironmentCustomResources(upload func(string, ...Uploadable) (string, error)) ([]string, error) {
+	return t.uploadCustomResources(upload, envCustomResourceFiles)
+}
+
+func (t *Template) uploadCustomResources(upload func(string, ...Uploadable) (string, error), fileNames []string) ([]string, error) {
+	urls := []string{}
+	for _, name := range fileNames {
+		url, err := t.uploadCompressedFile(upload, compressedFile{
+			name: path.Join(scriptDirName, name),
+			uploadables: []Uploadable{
+				{
+					name: customResourceZippedScriptName,
+					path: path.Join(customResourceRootPath, fmt.Sprintf("%s.js", name)),
+				},
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+		urls = append(urls, url)
+	}
+	return urls, nil
+}
+
+func (t *Template) uploadCompressedFile(upload func(string, ...Uploadable) (string, error), file compressedFile) (string, error) {
+	contents := []byte{}
+	for _, uploadable := range file.uploadables {
+		content, err := t.Read(uploadable.path)
+		if err != nil {
+			return "", err
+		}
+		uploadable.content = content.Bytes()
+		contents = append(contents, uploadable.content...)
+	}
+	// Use the SHA256 so that we'll upload every time when the file changes.
```

Comment:
```suggestion
	// Suffix with a SHA256 checksum of the compressedFile so that
	// only new content gets a new URL. Otherwise, if two compressedFiles have the
	// same content then the URL generated will be identical.
```